decavision/model_testing/testing.py

```diff
@@ -253,3 +256,198 @@ def generate_classification_report(self, path):
         print('Labels loaded')
         # Show classification report
         print(classification_report(cls_true, cls_pred, target_names=labels, digits=4))
+
+
+
+class ModelTesterMultilabel:
+    """
+    Class to use a trained multilabel image classification model on some images. Using this
+    when working with a TPU disables eager execution.
+
+    Arguments:
+        model (str): path to trained model
+        json_file (str): path to json file containing image ids and their associated labels
+        categories (list[str]): list of potential categories that the model can return
+    """    
+    
+    def __init__(self, model, json_file, categories):
+        use_tpu, use_gpu = utils.check_PU()
+        if use_tpu:
+            # necessary because keras generators don'T work with TPUs...
+            tf.compat.v1.disable_eager_execution()
+        try:
+            self.model = load_model(model, custom_objects={"_f1_score": f1_score, "f1_score": f1_score})
+            # efficientnets have the scaling included in them so no need to rescale the images when loading
+            if self.model.name[0] in ['B', 'V']:
+                self.rescaling = 1
+            else:
+                self.rescaling = 255
+            print('Model loaded correctly')
+        except Exception as e:
+            print('There was a problem when trying to load your model: {}'.format(e))
+
+        data = []
+        with open(json_file, 'r') as file:
+            values = json.load(file)
+            for f, l in values.items():
+                data.append({"filenames": f, "labels": l})
+
+        self.input_shape = self.model.input_shape[1:3]
+        self.df = pd.DataFrame(data)
+        self.values = values
+        self.json_file = json_file
+        self.categories = categories
+        print("Model name: ",self.model.name)
+
+
+    def _load_dataset(self, path):
+        """
+        Load dataset into a keras generator. Images must be contained in single
+        folder for each class. A dataframe is required with 2 columns: original 
```

Comment:
At many places in the docstrings it says that the images must be in a single folder for each class. The `for each class` part doesn't apply in this case.

Reply:
done.